src/AzureAD.tsx

```diff
@@ -45,66 +43,77 @@ export interface IAzureADProps {
   forceLogin?: boolean
 }
 
-interface IState {
-  authenticationState: AuthenticationState,
+interface IAzureADState {
+  authenticationState: AuthenticationState
 }
 
-class AzureAD extends React.Component<IAzureADProps, IState> {
-  private authProvider: MsalAuthProvider;
+class AzureAD extends React.Component<IAzureADProps, IAzureADState> {
+  private authProvider: MsalAuthProvider = this.props.provider.getAuthProvider();
+
+  // tslint:disable-next-line: member-ordering
+  public state: Readonly<IAzureADState> = {
+    authenticationState: this.authProvider.authenticationState
+  };
 
   constructor(props: IAzureADProps) {
     super(props);
 
-    this.authProvider = this.props.provider.getAuthProvider();
-    this.authProvider.onAuthenticationStateChanged = this.updateAuthenticationState;
+    this.authProvider.onAuthenticationStateChanged = this.setAuthenticationState;
+    this.authProvider.onAccountInfoChanged = this.onAccountInfoChanged;
 
-    const authState = this.authProvider.authenticationState;
-    this.state = { authenticationState: authState };
+    if (props.reduxStore) {
+      this.authProvider.registerReduxStore(props.reduxStore);
+    }
 
-    if (authState === AuthenticationState.Authenticated) {
-      const account = this.authProvider.getAccountInfo();
-      this.updateAuthenticationState(authState, account);
-    } else if (this.state.authenticationState === AuthenticationState.Unauthenticated && this.props.forceLogin) {
-      this.login();
+    const { authenticationState } = this.state;
+    if (authenticationState === AuthenticationState.Authenticated) {
+      const accountInfo = this.authProvider.getAccountInfo();
+      if (accountInfo && props.accountInfoCallback) {
+        this.onAccountInfoChanged(accountInfo);
+      }
+    } else if (authenticationState === AuthenticationState.Unauthenticated) {
+      if (props.forceLogin) {
+        this.login();
+      }
     }
   }
 
   public render() {
+    const { authenticatedFunction, unauthenticatedFunction, children } = this.props;
     switch (this.state.authenticationState) {
       case AuthenticationState.Authenticated:
-        if (this.props.authenticatedFunction) {
-          return this.props.authenticatedFunction(this.logout) || null;
+        if (authenticatedFunction) {
+          return authenticatedFunction(this.logout) || children;
```

Comment:
If the authenticatedFunction() is set, but does not return any elements, still allow the children to be rendered.